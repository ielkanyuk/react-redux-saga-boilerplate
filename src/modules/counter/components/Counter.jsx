import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export class Counter extends React.Component {
    static propTypes = {
        increment: PropTypes.func.isRequired,
        decrement: PropTypes.func.isRequired,
        number: PropTypes.number,
    };

    handleIncrement = () => {
        const { increment } = this.props;
        increment();
    };

    handleDecrement = () => {
        const { decrement } = this.props;
        decrement();
    };

    render() {
        const { number } = this.props;
        return (
            <div>
                <div className="form-buttons">
                    <span className="info info_success">{number}</span>
                    <button type="button" className="button small" onClick={this.handleIncrement}>+</button>
                    <button type="button" className="button small" onClick={this.handleDecrement}>-</button>
                </div>
                <div>
                    <Link to="/currency">Go to currency</Link>
                </div>
            </div>
        );
    }
}