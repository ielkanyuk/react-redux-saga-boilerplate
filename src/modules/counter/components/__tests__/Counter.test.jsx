/* eslint-env jest/globals */
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Counter } from '../Counter';

it('renders correctly', () => {
    const tree = renderer.create(
        <MemoryRouter>
            <Counter
                decrement={() => {}}
                increment={() => {}}
            />
        </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});