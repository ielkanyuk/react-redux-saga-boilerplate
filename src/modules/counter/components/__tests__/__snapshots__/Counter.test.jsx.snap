// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="form-buttons"
  >
    <span
      className="info info_success"
    />
    <button
      className="button small"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="button small"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div>
    <a
      href="/currency"
      onClick={[Function]}
    >
      Go to currency
    </a>
  </div>
</div>
`;
